src/node/internal/internal_http_incoming.ts

```diff
@@ -67,6 +75,39 @@ export class IncomingMessage extends Readable implements _IncomingMessage {
     ): void => {
       incoming.#setFetchResponse(response);
     };
+
+    // This method sets the socket property of the IncomingMessage object.
+    // Please rest assured that this method implements a subset of Socket since
+    // in Node.js it's net.Socket which isn't possible to implement within our own
+    // implementation since our implementation is based on Request and Response objects.
+    setIncomingMessageSocket = (
+      incoming: IncomingMessage,
+      { headers, localPort }: { headers: Headers; localPort: number }
+    ): void => {
+      incoming.#socket = {
+        encrypted: headers.get('x-forwarded-proto') === 'https',
+        get remoteAddress(): string {
+          // This is defined in production, and will fallback to localhost on local development
+          // where request headers does not contain cf-connecting-ip.
+          return headers.get('cf-connecting-ip') ?? '127.0.0.1';
+        },
+        get remotePort(): number {
+          // We return 80 by default.
+          return 80;
```

Comment:
80 feels weird.
all modern operating systems only generate ports in the range of 2^15 and 2^16 for the ephemeral source ports on sockets.

Reply:
e.g. just grab `32768` for a realistic local port number